### PR TITLE
New package: Microsoft.VisualStudio.2010.ToolsOfficeRuntime version 10.0.60828

### PR DESCRIPTION
**Pull request by bkazoo**
###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/102129)

### CODE REVIEW

**Comment by bkazoo** [comment]
@microsoft-github-policy-service agree

File: manifests/m/Microsoft/VisualStudio/2010/ToolsOfficeRuntime/60828.00/Microsoft.VisualStudio.2010.ToolsOfficeRuntime.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+PackageVersion: '60828.00'
+InstallerLocale: en-US
+InstallerType: exe
+InstallerSwitches:
+  Silent: /q
+  SilentWithProgress: /q
+  Custom: /norestart
+UpgradeBehavior: install
+Installers:
+- Architecture: x86
```
Comment (Trenly) [inline manifests/m/Microsoft/VisualStudio/2010/ToolsOfficeRuntime/60828.00/Microsoft.VisualStudio.2010.ToolsOfficeRuntime.installer.yaml:14]: Please also add the x64 architecture node
```
Found Visual Studio 2010 Tools for Office Runtime [Microsoft.VisualStudio.2010.ToolsOfficeRuntime] Version 60828.00
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://download.microsoft.com/download/C/A/8/CA86DFA0-81F3-4568-875A-7E7A598D4C1C/vstor_redist.exe
  ██████████████████████████████  38.5 MB / 38.5 MB
Successfully verified installer hash
Starting package install...
Successfully installed

--> Refreshing environment variables

--> Comparing ARP Entries

DisplayName                                                 DisplayVersion Publisher             ProductCode
-----------                                                 -------------- ---------             -----------
Microsoft Visual Studio 2010 Tools for Office Runtime (x64) 10.0.60828     Microsoft Corporation Microsoft Visual Studio 2010 Tools for Office Run...
Microsoft Visual C++ 2010  x64 Redistributable - 10.0.40219 10.0.40219     Microsoft Corporation {1D8E6291-B0D5-35EC-8441-6616F567A0F7}
Microsoft Visual Studio 2010 Tools for Office Runtime (x64) 10.0.60833     Microsoft Corporation {47C2CCDB-7A04-3797-992B-A84D3E90258F}
Microsoft Visual C++ 2010  x86 Redistributable - 10.0.40219 10.0.40219     Microsoft Corporation {F0C3E5D1-1ADE-321E-8167-68EF0DE699A5}
```

File: manifests/m/Microsoft/VisualStudio/2010/ToolsOfficeRuntime/60828.00/Microsoft.VisualStudio.2010.ToolsOfficeRuntime.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Created with Komac v1.4.1
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.4.0.schema.json
+
+PackageIdentifier: Microsoft.VisualStudio.2010.ToolsOfficeRuntime
+PackageVersion: '60828.00'
```
Comment (Trenly) [inline manifests/m/Microsoft/VisualStudio/2010/ToolsOfficeRuntime/60828.00/Microsoft.VisualStudio.2010.ToolsOfficeRuntime.installer.yaml:5]:
```suggestion
PackageVersion: '10.0.60833'
```
This should match what is written to Control Panel

File: manifests/m/Microsoft/VisualStudio/2010/ToolsOfficeRuntime/60828.00/Microsoft.VisualStudio.2010.ToolsOfficeRuntime.locale.en-US.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Created with Komac v1.4.1
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.4.0.schema.json
+
+PackageIdentifier: Microsoft.VisualStudio.2010.ToolsOfficeRuntime
+PackageVersion: '60828.00'
```
Comment (Trenly) [inline manifests/m/Microsoft/VisualStudio/2010/ToolsOfficeRuntime/60828.00/Microsoft.VisualStudio.2010.ToolsOfficeRuntime.locale.en-US.yaml:5]:
```suggestion
PackageVersion: '10.0.60833'
```

File: manifests/m/Microsoft/VisualStudio/2010/ToolsOfficeRuntime/60828.00/Microsoft.VisualStudio.2010.ToolsOfficeRuntime.locale.en-US.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Created with Komac v1.4.1
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.4.0.schema.json
+
+PackageIdentifier: Microsoft.VisualStudio.2010.ToolsOfficeRuntime
+PackageVersion: '60828.00'
+PackageLocale: en-US
+Publisher: Microsoft Corporation
+PublisherUrl: https://www.microsoft.com
+PackageName: Visual Studio 2010 Tools for Office Runtime
```
Comment (Trenly) [inline manifests/m/Microsoft/VisualStudio/2010/ToolsOfficeRuntime/60828.00/Microsoft.VisualStudio.2010.ToolsOfficeRuntime.locale.en-US.yaml:9]:
```suggestion
PackageName: Microsoft Visual Studio 2010 Tools for Office Runtime
```
This should match control panel as closely as possible

File: manifests/m/Microsoft/VisualStudio/2010/ToolsOfficeRuntime/60828.00/Microsoft.VisualStudio.2010.ToolsOfficeRuntime.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Created with Komac v1.4.1
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.4.0.schema.json
+
+PackageIdentifier: Microsoft.VisualStudio.2010.ToolsOfficeRuntime
+PackageVersion: '60828.00'
```
Comment (Trenly) [inline manifests/m/Microsoft/VisualStudio/2010/ToolsOfficeRuntime/60828.00/Microsoft.VisualStudio.2010.ToolsOfficeRuntime.yaml:5]:
```suggestion
PackageVersion: '10.0.60833'
```

**Review by Trenly — CHANGES_REQUESTED** [review]
You will need to add the `ProductCode` field for each installer since the installer adds multiple components.

You should also add the primary `AppsAndFeaturesEntries` for each as the architecture being in the name may cause issues if `AppsAndFeaturesEntries` are omitted.

File: manifests/m/Microsoft/VisualStudio/2010/ToolsOfficeRuntime/60828.00/Microsoft.VisualStudio.2010.ToolsOfficeRuntime.yaml (hunk 5 above)
Comment (bkazoo) [reply manifests/m/Microsoft/VisualStudio/2010/ToolsOfficeRuntime/60828.00/Microsoft.VisualStudio.2010.ToolsOfficeRuntime.yaml]: Control Panel shows **10.0.60828** for both x86 and x64 architectures, so I am using this for PackageVersion.

'PS> Get-WmiObject Win32_Product' returns **10.0.60833**, which I am using on each architecture DisplayVersion:

**x64:**
IdentifyingNumber : {47C2CCDB-7A04-3797-992B-A84D3E90258F}
Name              : Microsoft Visual Studio 2010 Tools for Office Runtime (x64)
Vendor            : Microsoft Corporation
**Version           : 10.0.60833**
Caption           : Microsoft Visual Studio 2010 Tools for Office Runtime (x64)


**x86:**
IdentifyingNumber : {39ABE1E5-A84D-3FC5-8311-776474862F50}
Name              : Microsoft Visual Studio 2010 Tools for Office Runtime (x86)
Vendor            : Microsoft Corporation
**Version           : 10.0.60833**
Caption           : Microsoft Visual Studio 2010 Tools for Office Runtime (x86)

**Comment by bkazoo** [comment]
x64:
x64_version

x64_productcode

x86:
x86_version

x86_productcode

**Comment by stephengillie** [comment]
Hi @bkazoo,

Visual Studio packages need approval, and I believe this is included. 

@edwardskrod, may I ask for your input?

**Comment by stephengillie** [comment]
> Hello @bkazoo,
> 
> The package manager bot determined that the metadata was not compliant.
> 
> Please verify the manifest file is compliant with the package manager [1.4 manifest specification](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0). Make sure the ID is of the form publisher.appname and that the folder structure is manifests\partition\publisher\appname\version. Note: The path and "PackageIdentifier" are case sensitive. Be sure to use a tool like [VSCode](https://code.visualstudio.com/) to make sure the manifest YAML syntax is correct.
> 
> You could also try our [Windows Package Manager Manifest Creator](https://github.com/microsoft/winget-create) or the [YamlCreate script](https://github.com/microsoft/winget-pkgs/blob/master/doc/tools/YamlCreate.md).
> 
> For details on the specific error, see the details link below in the build pipeline.
> 
> Template: msftbot/validationError/manifest/metadata

I'm not seeing the cause for this error. This package installs normally in a Windows 11 VM.

**Comment by bkazoo** [comment]
> > Hello @bkazoo,
> > 
> > The package manager bot determined that the metadata was not compliant.
> > 
> > Please verify the manifest file is compliant with the package manager [1.4 manifest specification](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0). Make sure the ID is of the form publisher.appname and that the folder structure is manifests\partition\publisher\appname\version. Note: The path and "PackageIdentifier" are case sensitive. Be sure to use a tool like [VSCode](https://code.visualstudio.com/) to make sure the manifest YAML syntax is correct.
> > 
> > You could also try our [Windows Package Manager Manifest Creator](https://github.com/microsoft/winget-create) or the [YamlCreate script](https://github.com/microsoft/winget-pkgs/blob/master/doc/tools/YamlCreate.md).
> > 
> > For details on the specific error, see the details link below in the build pipeline.
> > 
> > Template: msftbot/validationError/manifest/metadata
> 
> I'm not seeing the cause for this error. This package installs normally in a Windows 11 VM. 

I updated the folder structure after changing the version to resolve this, so I hope that this is no longer an issue.

**Comment by stephengillie** [comment]
> I updated the folder structure after changing the version to resolve this, so I hope that this is no longer an issue.

Yes, it's no longer an issue. This can be determined by seeing both green labels `Validation-Completed` and `Azure-Pipeline-Passed` labeles affixed by Wingetbot.

**Comment by stephengillie** [comment]
Hi @edwrdskrod,

Is this package okay to add?

**Comment by bkazoo** [comment]
Are there any pending changes that should be made?

**Comment by Trenly** [comment]
cc @edwardskrod @denelon

**Comment by ItzLevvie** [comment]
[Policy] Area-External

**Comment by ItzLevvie** [comment]
[Policy] Blocking-Issue

**Comment by bkazoo** [comment]
@Trenly are there any changes required?

**Comment by bkazoo** [comment]
Who could review this?

**Comment by denelon** [comment]
@bkazoo is this a "net new" package for WinGet?

**Review by mdanish-kh — CHANGES_REQUESTED** [review]
Looks like the Package already exists under [Microsoft.VSTOR](https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Microsoft/VSTOR) package identifier. Is this the only version of the package that is published? If yes, then I don't think that we should be adding this under a new ID and removing the previous since that would be a breaking change.

If no, then the current package ID format feels more like a logical home for these packages. We can add other versions  in a similar fashion (`Microsoft.VisualStudio.2019.ToolsOfficeRuntime`, `Microsoft.VisualStudio.2022.ToolsOfficeRuntime` etc..), and remove the other package ID since there's more value here now doing things this way

**Comment by mdanish-kh** [comment]
A quick google search tells me that this may be the only version that's published. I don't see a similar package for Visual Studio 2019, 2022 etc..

I think we should stick to the `Microsoft.VSTOR` package identifier....

**Comment by mdanish-kh** [comment]
cc @khaffner as the original contributor of Microsoft.VSTOR

**Comment by Trenly** [comment]
> I think we should stick to the `Microsoft.VSTOR` package identifier....

I agree; It makes it clear that its not part of the Visual Studio releases

**Comment by bkazoo** [comment]
Sure, let's keep the other package as it seems to be the same and it is already submitted.
Duplicated work could have been avoided as we both submitted the same package (this has been waiting with no changes requested since April).

**Comment by Trenly** [comment]
> Duplicated work could have been avoided as we both submitted the same package (this has been waiting with no changes requested since April).

Certainly there was some misunderstanding. I'll add a note to my tracker to review PRs marked as "blocked" more often

* Resolved by #109111

**Pull request closed without merging**